pkg/lib/synthesize/non_inferrable_csv.go

```diff
@@ -0,0 +1,31 @@
+package synthesize
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/api/pkg/lib/version"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+type RegistryV2CSV struct {
```

Comment:
Feels like this shouldn't live here, instead, should be in operator-framework/api.

Reply:
you are right. The decision to put it here is that it makes development faster. This was discussed very early on before i started working on it. The decision is that we put it here first and move it along with others once we are refactoring the API repo.

Reply:
>The question on my mind is whether the internal representation is necessary. Could we get away with directly using the CSV type from the api repo?

That was what I was trying to hint at, but didn't explicitly mention it as I haven't studied the CSV type to be confident that'll do the job. But if we're confident that's possible we should probably reuse that type.

Reply:
I misunderstood what this was because of the name. This is essentially the `olm.yaml` described in the enhancement. In that case, it makes sense for this to be here.